Snakefile

```diff
@@ -0,0 +1,402 @@
+            mv {input.nextclade_info} {output.nextclade_info}
+        else
+            ./bin/run-nextclade \
+                {input.sequences:q} \
+                {params.new_info} \
+                {params.nextclade_input_dir} \
+                {params.nextclade_output_dir}
+            # Join new and old clades, so that next run won't need to process sequences that are already processed
+            ./bin/join-rows \
+                {input.nextclade_info:q} \
+                {params.new_info} \
+                -o {output.nextclade_info:q}
+        fi
+        """
+
+rule generate_metadata:
+    input:
+        existing_metadata = "data/{database}/metadata_transformed.tsv",
+        new_metadata = "data/{database}/nextclade.tsv",
+    output:
+        metadata = "data/{database}/metadata.tsv"
+    # note: the shell scripts which predated this snakemake workflow
+    # overwrote the existing_metadata here
+    shell:
+        """
+        ./bin/join-metadata-and-clades \
+            {input.existing_metadata} \
+            {input.new_metadata} \
+            -o {output.metadata}
+        """
+
+rule flag_metadata:
+    ### only applicable for GISAID
+    input:
+        metadata = "data/gisaid/metadata.tsv"
+    output:
+        metadata = "data/gisaid/flagged_metadata.txt"
+    shell:
+        """
+        ./bin/flag-metadata {input.metadata} > {output.metadata}
+        """
+
+rule check_locations:
+    input:
+        metadata = "data/{database}/metadata.tsv"
+    params:
+        unique_id = lambda w: "gisaid_epi_isl" if w.database=="gisaid" else "genbank_accession"
+    output:
+        location_hierarchy = "data/{database}/location_hierarchy.tsv"
+    shell:
+        """
+        ./bin/check-locations {input.metadata} {output.location_hierarchy} {params.unique_id}
+        """
+
+rule notify_gisaid:
+    input:
+        flagged_annotations = rules.transform_gisaid_data.output.flagged_annotations,
+        # metadata = "data/gisaid/metadata.tsv",
+        additional_info = "data/gisaid/additional_info.tsv",
+        flagged_metadata = "data/gisaid/flagged_metadata.txt",
+        location_hierarchy = "data/gisaid/location_hierarchy.tsv"
+    params:
+        s3_bucket = config["s3_src"]
+    output:
+        touch("data/gisaid/notify.done")
+    run:
+        shell("./bin/notify-slack --upload flagged-annotations < {input.flagged_annotations}")
+        # notify-on-metadata-change disabled as csv-diff runs out of memory
+        # shell("./bin/notify-on-metadata-change {input.metadata} {params.s3_bucket}/metadata.tsv.gz gisaid_epi_isl")
+        shell("./bin/notify-on-additional-info-change {input.additional_info} {params.s3_bucket}/additional_info.tsv.gz")
+        shell("./bin/notify-on-flagged-metadata-change {input.flagged_metadata}  {params.s3_bucket}/flagged_metadata.txt.gz")
+        shell("./bin/notify-on-location-hierarchy-addition {input.location_hierarchy} source-data/location_hierarchy.tsv")
```

Comment:
Why not make this one `shell:` block instead of a `run:` block with lots of `shell()`s?

Reply:
I think they're much of a muchness. I chose this form as it makes it easier to break those scripts out into python commands if desired down the track. This is also noted in #239 so I'll resolve this for now.